docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md

```diff
@@ -806,6 +806,167 @@ One challenge when developing locally is that a developer might make modificatio
+      node: entryNode,
+      updatedAtUTC: updatedAt,
+    })
+  }
+}
+```
+
+##### Warn if no support
+
+Lastly we'll want to give our users a good experience and give a warning if they're using a version of Gatsby that does not support Content Sync
+
+```javascript:title=source-plugin/utils.js
+// highlight-start
+let warnOnceForNoSupport = false
+// highlight-end
+
+export function createNodeManifest({
+  // ...
+}) {
+  // ...
+
+  if (shouldCreateNodeManifest) {
+    // ...
+  } else if (
+    // highlight-start
+    // it's helpful to let users know if they're using an outdated Gatsby version so they'll upgrade for the best experience
+    isPreview && !createNodeManifestIsSupported && !warnOnceForNoSupport
+  ) {
+    console.warn(
+      `${sourcePluginName}: Your version of Gatsby core doesn't support Content Sync (via the unstable_createNodeManifest action). Please upgrade to the latest version to use Content Sync in your site.`
+    )
+    // This is getting called for every entry node so we don't want the console logs to get cluttered
+    warnOnceForNoSupport = true
+    // highlight-end
+  }
+}
+```
+
+#### Content Sync data source (or CMS) changes
+
+The CMS will need to send a webhook when content is changed
+
+##### Using the gatsby preview extension
+
+This package is currently in the process of being open sourced and should be available shortly. For context, here is what the package provides, a button to handle opening the Content Sync waiting room. You will need to create a button similar to this for the time being.
+![Open Preview Button Screenshot](https://user-images.githubusercontent.com/18426780/61498855-37e35a00-a982-11e9-8201-ab96be74f1f1.png)
+
+##### Configuration
+
+You will need to store the Content Sync URL from a given Gatsby Cloud site. This will look something like `https://gatsbyjs.com/content-sync/<siteId>`. This is often done in the plugin extension or app extension configuration, this will differ from CMS to CMS depending on how they handle their plugin ecosystem.
+
+##### Building up the manifest id
+
+Recall that we need to create a matching manifest id in the CMS AND the Gatsby plugin. Whenever content is saved we can build up a new manifest id that will look the same as the manifest id we created in the source plugin
+
+```javascript
+const manifestId = `${project.id}-${entryItem.id}-${updatedAt}`
+```
+
+In the CMS we should have access to the project id (if the CMS uses one) and the content id, the only thing we need to figure out how to grab is the timestamp that the content was updated at.
+
+##### Opening the Content Sync waiting room
+
+When we've built up the manifest id, when the "Open Preview" button is clicked, we need to open a new window to the Content Sync waiting room as follows `https://gatsbyjs.com/content-sync/<siteId>/<sourcePluginName>/<manifestId>`. This will ensure that this piece of content is linked to the corresponding content that was created in their Gatsby site.
```

Comment:
```suggestion
Once we've built a `manifestId` and `POST`ed to the preview build webhook, we need to open a new tab/window with a modified version of the Content Sync URL. You get that by grabbing the Content Sync URL you stored in the CMS plugin earlier and appending the Gatsby source plugin name and content's `manifestId` that you just created. Like so...
`https://gatsbyjs.com/content-sync/<siteId>/<sourcePluginName>/<manifestId>`
```

I'm mainly try to call out the connection between this url and the stored content sync url from an earlier step with this suggestion